File: backend/app/models/change_log.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db import Base


class PokemonChangeLog(Base):
    """Append-only record of detected PokeAPI field changes. Public change-log evidence trail."""

    __tablename__ = "pokemon_change_log"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    pokemon_id: Mapped[int] = mapped_column(ForeignKey("pokemon.id"), nullable=False)
    field_name: Mapped[str] = mapped_column(String, nullable=False)
    old_value: Mapped[str] = mapped_column(String, nullable=False)
    new_value: Mapped[str] = mapped_column(String, nullable=False)
    detected_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

File: frontend/src/types/changeLog.ts
export interface ChangeLogEntry {
  id: number;
  pokemon_id: number;
  field_name: string;
  old_value: string;
  new_value: string;
  detected_at: string;
}

export interface MoveChangeLogEntry {
  id: number;
  move_id: number;
  field_name: string;
  old_value: string;
  new_value: string;
  detected_at: string;
}
